### PR TITLE
Support Dockerfile-specific ignore-file with watch

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Follow https://docs.docker.com/reference/dockerfile/#dockerignore-file and first consider a Dockerfile-specific ignore file when computing exclude patterns for watch. 

**Related issue**
closes https://github.com/docker/compose/issues/12191

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

### CODE REVIEW

**Review by glours — APPROVED** [review]
👌 LGTM

File: pkg/watch/dockerignore.go [hunk 1]
```diff
@@ -61,13 +63,28 @@ func (i dockerPathMatcher) MatchesEntireDir(f string) (bool, error) {
...
+func LoadDockerIgnore(build *types.BuildConfig) (*dockerPathMatcher, error) {
+	repoRoot := build.Context
 	absRoot, err := filepath.Abs(repoRoot)
 	if err != nil {
 		return nil, err
 	}
 
-	patterns, err := readDockerignorePatterns(absRoot)
+	// first try Dockerfile-specific ignore-file
+	f, err := os.Open(filepath.Join(repoRoot, build.Dockerfile+".dockerignore"))
```
Comment (thaJeztah) [inline pkg/watch/dockerignore.go:74]: Is `build.Dockerfile` always just the file name, or can it be an absolute path? (in that case, no need to join if with the repo-root)

Comment (ndeloof) [reply pkg/watch/dockerignore.go:74]: I'm actually running some tests as I wondered watch do abspath resolution, while compose-go should already have done

**Comment by thaJeztah** [comment]
Quick blurb; not sure if "watch" is supported for windows container (and the classic builder), but I think the classic (non-buildkit) doesn't support per-dockerfile ignore-files; just in case that's relevant for this use 🤗

File: pkg/watch/dockerignore.go (hunk 1 above)
Comment (ndeloof) [reply pkg/watch/dockerignore.go:74]: `Dockerfile` is always just the file name within the context

**Pull request merged**